### ISSUE TITLE
Full tests fail

### ISSUE THREAD

**Issue by rogerbinns** [utelle/apsw-sqlite3mc#5]
The mctests run in isolation run fine.  However when run at the end of the APSW test suite there is an Corrupt Error returned.

A write to the database succeeds, but then the read gets that error.

**Comment by rogerbinns** [comment]
This is a [bug in sqlite3 multiple ciphers](https://github.com/utelle/SQLite3MultipleCiphers/issues/156)

**Comment by utelle** [comment]
> The mctests run in isolation run fine. However when run at the end of the APSW test suite there is an Corrupt Error returned.

At the moment I don't know why this error occurs. Does the APSW test suite make use of non-default config settings? Does it add a different VFS or change the default VFS?

> A write to the database succeeds, but then the read gets that error.

Smells like the read buffer is somehow corrupted. But why? In https://github.com/utelle/SQLite3MultipleCiphers/issues/156 you mention the use of `SQLITE_CONFIG_MMAP_SIZE`. Is this parameter also modified in the APSW test suite? AFAIK memory mapped files are not used, if encryption is enabled.

Hopefully I will be able to reproduce the error in my development environment.

**Comment by rogerbinns** [comment]
The APSW test suite exercises every API and object in SQLite, multiple times.  That includes VFS functionality, changing settings etc.  The change I made in this project is to have the APSW test do [this very small test](https://github.com/utelle/apsw-sqlite3mc/blob/sqlite3mc/apsw/mctests.py) after having run all the APSW stuff.

I got the corrupt error, but not if I ran the mctests solo.  Then I reduced things to the smallest cause which was the [MMAP issue](https://github.com/utelle/SQLite3MultipleCiphers/issues/156) which does not have anything to do with the APSW test suite.  I'll continue this over on that issue.
